### DISCUSSION TITLE
Non Maintained next-routes Library

### DISCUSSION THREAD

**Discussion by SudharakaP** [opencollective/opencollective#3789] [Ideas]
@opencollective/core : Just an idea, I am not sure if this warrants a new issue or not (or whether if it was discussed before); but I think maybe we should think about migrating away from next-routes library as [it doesn't seem to be actively maintained anymore](https://github.com/fridays/next-routes/issues/334). If this is something of use I am willing to help out on this. 

Also as a side note, it seems interesting that the [out of memory exception we are getting](https://github.com/vercel/next.js/issues/15855) seems to only occur when we move between some pages (let me know if the behavior is different for you). So maybe I guess this could improve things on that front as well. 🤔

**Comment by Betree** [comment]
It's indeed in the plans to have a try with the latest NextJS routes in the future. `next-routes` was used at the time because the support for this in NextJS was still a bit poor, but we may be able to migrate now. The main issue that we faced when we first looked at this was that we have some pretty complex routes with regexs, that can't be represented just by naming files in the file system (see [this one](https://github.com/opencollective/opencollective-frontend/blob/master/server/pages.js#L128) for example).

But since, NextJS introduced [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites), which support regexs in URL patterns. So we may be able to transition using that.

For the migration strategy, the ideal solution would be to progressively migrate by updating `next.config.js`, eventually moving the files in `pages` when it is non-breaking. It would be easy to test on our Zeit pre-deploys.

Feel free to create an issue, this is definitely something we want to look into :smiley:
